cmd/meroxa/root/apps/deploy.go

```diff
@@ -524,6 +525,62 @@ func (d *Deploy) validate(ctx context.Context) error {
 	return turbineCLI.ValidateBranch(ctx, d.logger, d.path)
 }
 
+func (d *Deploy) checkResourceAvailability(ctx context.Context) error {
+	d.logger.Infof(ctx, "Checking resource availability for application %q (%s) before deployment...", d.appName, d.lang)
+
+	var resourceNames []string
+	var err error
+
+	switch d.lang {
+	case GoLang:
+		resourceNames, err = turbineGo.GetResourceNames(ctx, d.logger, d.path, d.appName)
+	case JavaScript, NodeJs:
```

Comment:
I don't think we need to case for `NodeJs` since we standarize languages in the validation step: https://github.com/meroxa/cli/blob/master/cmd/meroxa/root/apps/deploy.go#L483.